alf/algorithms/data_transformer.py

```diff
@@ -862,4 +869,7 @@ def create_data_transformer(data_transformer_ctor, observation_spec):
     if len(data_transformer_ctor) == 1:
         return data_transformer_ctor[0](observation_spec)
 
+    if HindsightExperienceTransformer in data_transformer_ctor:
+        assert HindsightExperienceTransformer == data_transformer_ctor[0], \
+            "Hindsight relabeling should happen before all other transforms."
```

Comment:
The check here should be better merged with the FrameStacker check in `SequentialDataTransformer`

Reply:
Good point.  Removed here.